### PR TITLE
Aye voters drop page at Polkadot and Kusama asset hub

### LINKED ISSUE #6615: Aye voters drop page
image

## Main 
- handoff file updated - its in the same page as unlockable drop page - section aye voters 
- we are changing only the cover img, icon, we have to think about the name? Not sure about Aye voters exclusive mint

## Changes 
- cover img
- icon 

### Eligibility check 
- since there will be a list of addresses that are eligible for minting, we need to rework the action button section a bit 
- this will be shown for everyone that does not have connected wallet 
image
- once the user connects, its split into two states
   - user is not eligible - button changes to disabled state with small tooltips
image
   - user is eligible - he can then proceed to minting 
image

### Cover and pfp
image
- updated in the file as well

### PR DESCRIPTION
**Pull request by Jarsen136**
**Thank you for your contribution** to the [KodaDot - One Stop Shop for Polkadot NFTs](https://kodadot.xyz).

👇 __ Let's make a quick check before the contribution.

## PR Type

- [ ] Bugfix
- [x] Feature
- [ ] Refactoring

## Needs QA check

- @kodadot/qa-guild please review

## Context

- [x] Closes #6615 
- [ ] Requires deployment <snek/rubick/worker>

#### Did your issue had any of the "$" label on it?

- [x] My DOT address: [Payout](https://canary.kodadot.xyz/dot/transfer/?target=16SjUbGKSdjCdWTy3NNT3JxbRVGGqD4mwkHpc6BD9U2Rp29Z)

## Screenshot 📸

- [ ] My fix has changed UI

### <samp>🤖 Generated by Copilot at fa7ee3c</samp>

This pull request adds a new feature to display a vote drop collection on the drops page, which is a special collection that can only be minted by users who have voted yes on a specific referendum. It also refactors the existing unlockable collection components to be more reusable and robust, and enables the use of the Polkassembly API to fetch the voting data.

### <samp>🤖 Generated by Copilot at fa7ee3c</samp>

> _We're building a vote drop for the NFT crew_
> _With `UnlockableCollectionInfo` and `VoteDropContainer` too_
> _We'll fetch the votes from the Polkassembly API_
> _And mint the collection on the count of three_

### CODE REVIEW

**Comment by Jarsen136** [comment]
The mock page is here: /dot/drops/vote-drop

image


For now, it's still blocked by the work on the indexer side.

**Comment by exezbcz** [comment]
good! one detail, text in button: its eligibility 
image

**Comment by Jarsen136** [comment]
> > For now, it's still blocked by the work on the indexer side.
> 
> Sir maybe ping?

Oh, I just thought you are still developing the indexer for the drop.  

> 
> What needs to be done?

Id of the free-drop collection. And what network will the collection belong to? @vikiival

**Comment by vikiival** [comment]
> Oh, I just thought you are still developing the indexer for the drop.

Indexer was there for already for a week, no need to develop our own for this.

> re page

Even though I am not eligible I am able to mint.

Screenshot 2023-08-22 at 14 53 00

> Id of the free-drop collection. And what network will the collection belong to?

FOR TEST STMN 
99

FOR PROD I dont have collection yet

> the API to mint nft

```
code goes here
```

**Comment by Jarsen136** [comment]
mock page: /ahk/drops/vote-drop

image

**Comment by Jarsen136** [comment]
> > Id of the free-drop collection. And what network will the collection belong to?
> 
> FOR TEST STMN 99
> 
> FOR PROD I dont have collection yet
> 
> > the API to mint nft
> 
> [API](https://github.com/kodadot/nft-gallery/blob/3ddb7440b21edd6d9a2323885a0388d6b84fc6b7/composables/transaction/mintToken/transactionMintStatemine.ts#L64)
@vikiival 
There is one thing I'm not sure about. For the production version, are we going to make the minting process on the frontend side? If that's the case, the detection of mint permissions would only be done on the front-end. Hackers can directly call the API to create NFTs without going through the page.

In the previous implementation of our free mint, we used to request the backend server to perform minting and transferring.

**Comment by vikiival** [comment]
Cc @yangwao maybe do this as holderOf collection? 
I can predrop some art to all supporters

**Comment by vikiival** [comment]
> Hackers can directly call the API to create NFTs without going through the page

It still need to be implemented for future drops. 

Rest is describe in the comment above

**Comment by yangwao** [comment]
> maybe do this as holderOf collection?

Holder of which collection?


image

**Comment by vikiival** [comment]
> Holder of which collection?


As i said above:

I can predrop some art to all supporters

**Comment by Jarsen136** [comment]
> 
> @vikiival There is one thing I'm not sure about. For the production version, are we going to make the minting process on the frontend side? If that's the case, the detection of mint permissions would only be done on the front-end. Hackers can directly call the API on-chain to create NFTs without going through the page.
> 

I do not mean to send drops directly to all supporters. I'm thinking about providing an api for users to mint. When a legitimate user requests the interface, the backend service will determine if they are a supporter and send them an NFT. I hope my thoughts have not caused any misunderstanding.

**Comment by vikiival** [comment]
> I'm thinking about providing an api for users to mint. 

> When a legitimate user requests the interface, the backend service will determine if they are a supporter and send them an NFT.

Yes thats how it works now with current drops

> I hope my thoughts have not caused any misunderstanding.

**Comment by vikiival** [comment]
@Jarsen136 any blockers from my side on this?

**Comment by Jarsen136** [comment]
> @Jarsen136 any blockers from my side on this?

There are three steps that need to be done.
1. Prepare the collection for drop for production environment @yangwao 
2. Implement the api to mint nft, like we have implemented with current drops. @vikiival 
3. Apply the api to the mint button on the page.  @Jarsen136 

Then we could release this aye voters drop page.

**Comment by vikiival** [comment]
> Implement the api to mint nft, like we have implemented with current drops

It is the same api as for drops

**Comment by vikiival** [comment]
> but doesn't have nft-pallet whitelist/allowlist?

It has only 3 minting options:

- public 
- issuer
- holderOF

As mentioned before it will work same as current drops: issuer + internal whitelist

**Comment by vikiival** [comment]
> Thanks for the clarification. At least we need a new campaign name for the same api I bet.

Screenshot 2023-08-29 at 00 33 12

**Review by vikiival — COMMENTED** [review]
~

**Comment by vikiival** [comment]
```json
{
  "data": {
    "collectionEntityById": {
      "id": "99",
      "name": "AYE Waifu",
      "image": "ipfs://ipfs/bafybeibe2jccdqti4gsrllbhh6bwbmllaf6q3c7gwpi3jnlpmdwntfcn2a",
      "meta": {
        "id": "bafkreigb6fxq6bt6ysgqknu4rxrsrngv3lif2mek7vnvm65o3fpflxnidq",
        "name": "AYE Waifu",
        "image": "ipfs://ipfs/bafybeibe2jccdqti4gsrllbhh6bwbmllaf6q3c7gwpi3jnlpmdwntfcn2a"
      }
    }
  }
}

```

image is there so @Jarsen136 it is a bug

**Comment by Jarsen136** [comment]
> * image not loading: ❌
> 
> image
> 
> https://deploy-preview-6664--koda-canary.netlify.app/ahk/gallery/99-2
> 

There's the same bug on free-drop. I have created an issue for it https://github.com/kodadot/nft-gallery/issues/7077

**Comment by Jarsen136** [comment]
> > image not loading aswell: ❌
> 
> this is either my bad or indexer one



> ```json
> {
>   "data": {
>     "collectionEntityById": {
>       "id": "99",
>       "name": "AYE Waifu",
>       "image": "ipfs://ipfs/bafybeibe2jccdqti4gsrllbhh6bwbmllaf6q3c7gwpi3jnlpmdwntfcn2a",
>       "meta": {
>         "id": "bafkreigb6fxq6bt6ysgqknu4rxrsrngv3lif2mek7vnvm65o3fpflxnidq",
>         "name": "AYE Waifu",
>         "image": "ipfs://ipfs/bafybeibe2jccdqti4gsrllbhh6bwbmllaf6q3c7gwpi3jnlpmdwntfcn2a"
>       }
>     }
>   }
> }
> ```
> 
> image is there so @Jarsen136 it is a bug


It might be indexer one 👀  @vikiival 

ahp 
id: 2
https://squid.subsquid.io/speck/v/v2/graphql
image

**Comment by vikiival** [comment]
Waiting for @JustLuuuu for Metadata :)

**Comment by vikiival** [comment]
also @Jarsen136 It should mint same metadata as for collection

**Comment by vikiival** [comment]
Also it should show Collection image instead of waifus

Screenshot 2023-09-01 at 13 46 17

**Comment by vikiival** [comment]
Screenshot 2023-09-01 at 13 47 01

@Jarsen136 wen user clicks on claim he/she will get random waifu, which is incorrect -> It should get same metadata string as the `collection.metadata`

**Comment by Jarsen136** [comment]
> Screenshot 2023-09-01 at 13 47 01
> 
> @Jarsen136 wen user clicks on claim he/she will get random waifu, which is incorrect -> It should get same metadata string as the `collection.metadata`

Thanks. I will modify it later.

File: components/collection/voteDrop/DropContainer.vue [hunk 1]
```diff
@@ -252,6 +253,14 @@ const { data: collectionData, refetch } = useGraphql({
   },
 })
 
+watch(collectionData, async () => {
+  if (collectionData.value) {
+    imageList.value = [
+      await sanitizeIpfsUrl(collectionData.value?.collectionEntity.image),
```
Comment (vikiival) [inline components/collection/voteDrop/DropContainer.vue:259]: you dont need await here as sanitize is a sync fn

File: components/collection/voteDrop/DropContainer.vue [hunk 2]
```diff
@@ -277,7 +286,7 @@ const mintedPercent = computed(() => {
 
 const hasUserMinted = computed(
   () =>
-    Boolean(collectionData.value?.nftEntitiesConnection?.totalCount) ||
+    collectionData.value?.nftEntitiesConnection?.edges?.[0]?.node?.id ||
```
Comment (vikiival) [inline components/collection/voteDrop/DropContainer.vue:289]: why?

File: components/collection/voteDrop/DropContainer.vue [hunk 3]
```diff
@@ -304,11 +313,9 @@ const handleMint = async () => {
   }
   isLoading.value = true
 
-  const randomIndex = getRandomInt(imageList.value.length - 1)
-  const image = resultList.value.at(randomIndex).image
-
+  const imageHash = collectionData.value.collectionEntity.image
   const hash = await createUnlockableMetadata(
```
Comment (vikiival) [inline components/collection/voteDrop/DropContainer.vue:317]:
```
const hash = collectionData.value.collectionEntity.metadata
```

**Review by vikiival — COMMENTED** [review]
~

**Comment by prury** [comment]
Check Eligibility not triggering anything atm when wallet is disconnected

File: components/collection/voteDrop/DropContainer.vue (hunk 2 above)
Comment (Jarsen136) [reply components/collection/voteDrop/DropContainer.vue]: The nft id would be used here. I have renamed it
image

File: components/collection/voteDrop/DropContainer.vue (hunk 3 above)
Comment (Jarsen136) [reply components/collection/voteDrop/DropContainer.vue]: Thanks for pointing out.

**Comment by Jarsen136** [comment]
> Check Eligibility button not triggering anything atm when wallet is disconnected

Fixed

File: components/collection/voteDrop/DropContainer.vue (hunk 1 above)
Comment (Jarsen136) [reply components/collection/voteDrop/DropContainer.vue]: Updated

**Comment by Jarsen136** [comment]
`https://squid.subsquid.io/polkadot-polkassembly/graphql` The indexer is down, so it would not work to check the eligibility for now.

image

**Comment by vikiival** [comment]
> The indexer is down, so it would not work to check the eligibility for now

Noted will come up with some backup plan for now

**Comment by Jarsen136** [comment]
There is another issue related to the format of the metadata on collection.

The format of metadata is text instead of json, which is why the image could not load properly before.
https://image-beta.w.kodadot.xyz/ipfs/bafkreiduhiiovev346oqlhnmovxzsjwjhh2g5blxoxxnmqsfxttwxgnp6a

I have to write this fallback to make the image load.
```
 typeof _data === 'string' ? JSON.parse(_data) : _data
```

IMO it's valuable to find out why it has such a format. wdyt @vikiival 

ahp
id: 2
https://squid.subsquid.io/speck/v/v2/graphql

image

image

image

**Comment by vikiival** [comment]
> The format of metadata is text instead of json, which is why the image could not load properly before.
> https://image-beta.w.kodadot.xyz/ipfs/bafkreiduhiiovev346oqlhnmovxzsjwjhh2g5blxoxxnmqsfxttwxgnp6a

cc @preschian 👀

**Comment by vikiival** [comment]
Screenshot 2023-09-02 at 14 37 50
 

Api works again

**Comment by Jarsen136** [comment]
> There is another issue related to the format of the metadata on collection.
> 
> The format of metadata is text instead of json, which is why the image could not load properly before. https://image-beta.w.kodadot.xyz/ipfs/bafkreiduhiiovev346oqlhnmovxzsjwjhh2g5blxoxxnmqsfxttwxgnp6a
> 
> I have to write this fallback to make the image load.
> 
> ```
>  typeof _data === 'string' ? JSON.parse(_data) : _data
> ```
> 

related with https://github.com/kodadot/nft-gallery/pull/7102

**Comment by preschian** [comment]
> > The format of metadata is text instead of json, which is why the image could not load properly before.
> > [image-beta.w.kodadot.xyz/ipfs/bafkreiduhiiovev346oqlhnmovxzsjwjhh2g5blxoxxnmqsfxttwxgnp6a](https://image-beta.w.kodadot.xyz/ipfs/bafkreiduhiiovev346oqlhnmovxzsjwjhh2g5blxoxxnmqsfxttwxgnp6a)
> 
> cc @preschian 👀

I will trace it here https://github.com/kodadot/workers/issues/145

**Comment by preschian** [comment]
> > > The format of metadata is text instead of json, which is why the image could not load properly before.
> > > [image-beta.w.kodadot.xyz/ipfs/bafkreiduhiiovev346oqlhnmovxzsjwjhh2g5blxoxxnmqsfxttwxgnp6a](https://image-beta.w.kodadot.xyz/ipfs/bafkreiduhiiovev346oqlhnmovxzsjwjhh2g5blxoxxnmqsfxttwxgnp6a)
> > 
> > 
> > cc @preschian 👀
> 
> I will trace it here [kodadot/workers#145](https://github.com/kodadot/workers/issues/145)

found the root cause here https://github.com/kodadot/workers/pull/146

File: components/collection/voteDrop/DropContainer.vue [hunk 4]
```diff
@@ -0,0 +1,354 @@
...
+                      <feBlend
+                        mode="normal"
+                        in="SourceGraphic"
+                        in2="BackgroundImageFix"
+                        result="shape" />
+                      <feGaussianBlur
+                        stdDeviation="7.5"
+                        result="effect1_foregroundBlur_394_6126" />
+                    </filter>
+                  </defs>
```
Comment (roiLeo) [inline components/collection/voteDrop/DropContainer.vue:59]: create svg file?

File: components/collection/voteDrop/DropContainer.vue [hunk 5]
```diff
@@ -0,0 +1,354 @@
...
+            "Unlockable Content" to reveal the surprise. Follow the schedule so
+            you don't miss this!
+          </div>
+          <NeoButton
+            tag="a"
+            href="https://hello.kodadot.xyz/fandom-toolbox/audience-growth/unlockables"
+            target="_blank"
+            variant="secondary"
+            class="mt-2">
+            Learn More
```
Comment (roiLeo) [inline components/collection/voteDrop/DropContainer.vue:143]:
```suggestion
            {{ $t('helper.learnMore') }}
```

**Review by roiLeo — APPROVED** [review]
Tested on my side and wfm ✅ 

When I change account on `/ahp/drops/vote-drop`, I see

Screenshot 2023-09-06 at 12-06-12 Referendum Mint KodaDot - NFT Market Explorer

while It shouldn't be: (after reload)
Screenshot 2023-09-06 at 12-06-59 Referendum Mint KodaDot - NFT Market Explorer

let it roll 🪨 

note: You really love `<span>`

File: components/collection/voteDrop/DropContainer.vue (hunk 4 above)
Comment (vikiival) [reply components/collection/voteDrop/DropContainer.vue:59]: yup please ❤️

Comment (Jarsen136) [reply components/collection/voteDrop/DropContainer.vue]: Done

**Comment by Jarsen136** [comment]
> 
> When I change account on `/ahp/drops/vote-drop`, I see
> while It should be: (after reload)

✅  Fixed

File: components/collection/voteDrop/DropContainer.vue [hunk 6]
```diff
@@ -26,38 +26,8 @@
...
-                        in2="BackgroundImageFix"
-                        result="shape" />
-                      <feGaussianBlur
-                        stdDeviation="7.5"
-                        result="effect1_foregroundBlur_394_6126" />
-                    </filter>
-                  </defs>
-                </svg>
+                class="is-flex is-align-items-center">
+                <img src="/unlockable-pulse.svg" />
```
Comment (vikiival) [inline components/collection/voteDrop/DropContainer.vue]: should be placed in that special folder @Jarsen136

Comment (Jarsen136) [reply components/collection/voteDrop/DropContainer.vue]: done

**Comment by prury** [comment]
Second check then: 
Eligibility works ✔️ 
images loading ✔️ 
already have button redirecting correctly ✔️ 
https://deploy-preview-6664--koda-canary.netlify.app/ahp/gallery/2-3 ✔️ 
https://deploy-preview-6664--koda-canary.netlify.app/ahk/gallery/99-2 ✔️ 
view collection button works ✔️ 
twitter message redirecting to free-drop, if that's okay then let's go

**Comment by roiLeo** [comment]
> Any suggestion? Do you mean using more semantic elements like these `<h1> <h2> <section> `

simple `<div>` even when it's inline, idk why but `<span>` looks weird with our prettier setup

**Comment by Jarsen136** [comment]
> twitter message redirecting to free-drop, if that's okay then let's go

Thanks. I have fixed it.

**Comment by yangwao** [comment]
🥵
pay 110 usd

**Comment by yangwao** [comment]
😍 Perfect, I’ve sent the payout 
💵 $110 @ 4.23 USD/DOT ~ 26.005 $DOT
🧗 [16SjUbGKSdjCdWTy3NNT3JxbRVGGqD4mwkHpc6BD9U2Rp29Z](https://polkadot.subscan.io/account/16SjUbGKSdjCdWTy3NNT3JxbRVGGqD4mwkHpc6BD9U2Rp29Z)
🔗 [0x6ad3eb59d60eaff7085e9fa5c5b5b8548f0e88d009a4941ca4b63f838647c259](https://polkadot.subscan.io/extrinsic/0x6ad3eb59d60eaff7085e9fa5c5b5b8548f0e88d009a4941ca4b63f838647c259)

🪅 Let’s grab another issue and get rewarded! 
🪄 [github.com/kodadot/nft-gallery/issues](https://github.com/kodadot/nft-gallery/issues/)

**Pull request merged**